src/lda.jl

```diff
@@ -219,3 +193,104 @@ function _lda_whitening!{T<:AbstractFloat}(C::Matrix{T}, regcoef::T)
     return scale!(E.vectors, v)
 end
 
+#### SubspaceLDA
+
+# When the dimensionality is much higher than the number of samples,
+# it makes more sense to perform LDA on the space spanned by the
+# within-group scatter.
+
+immutable SubspaceLDA{T<:AbstractFloat}
+    projw::Matrix{T}
+    projLDA::Matrix{T}
+    λ::Vector{T}
+    cmeans::Matrix{T}
+    cweights::Vector{Int}
+end
+
+indim(M::SubspaceLDA) = size(M.projw,1)
+outdim(M::SubspaceLDA) = size(M.projLDA, 2)
+
+projection(M::SubspaceLDA) = M.projw*M.projLDA
+
+Base.mean(M::SubspaceLDA) = vec(sum(M.cmeans*Diagonal(M.cweights/sum(M.cweights)),2))
```

Comment:
Looks like you're spacing out operators in some places but not everywhere. I think it'd be nice from a readability standpoint to surround things like `*` and `/` with whitespace. Up to you though, just a suggestion.